content/BindingMethods/BindingMethods.q.md

```diff
@@ -0,0 +1,1601 @@
+
+- It binds the value `John` to the first positional placeholder `?` of the variable `first_name` in the SQL statement.
+
+- It binds the value `Smith` to the second positional placeholder `?` of the variable `last_name` in the SQL statement.
+
+- It replaces the value `John` to the first positional placeholder `?` of the variable `first_name` in the SQL statement.
+
+- It assigns the value `Smith` to the second positional placeholder `?` of the variable `last_name` in the SQL statement.
+
+- It replaces the value `Smith` to the second positional placeholder `?` of the variable `last_name` in the SQL statement.
+
+:::
+
+
+:::
+
+/// type=REPL, filename=[connection.php,student.php]
+
+```php
+// connection.php
+<?php
+    $host = 'localhost';
+    $db = 'LibraryDB';
+    $port = '5432';
+    $username = 'postgres';
+    $password = 'Admin01';
+
+    try {
+        $conn = new PDO("pgsql:host=$host;port=$port;dbname=$db;user=$username;password=$password");
+        if ($conn) {
+            echo "Successfully connected to the database. <br />";
+        }
+    } catch (Exception $e) {
+        echo "Unable to establish a connection."; 
+    }
+?>
+```
+
+```php
+// student.php
+<?php
+    require_once("connection.php");
+
+    $data = array(
+        'first_name' => 'John',
+        'last_name' => 'Smith'
+    );
+
+    $sql = "SELECT * FROM students WHERE first_name = :first_name and last_name = :last_name";
+
+    try {
+        $pstmt = $conn->prepare($sql);
+        $pstmt->bindValue(':first_name', $data['first_name'], PDO::PARAM_STR);
+        $pstmt->bindValue(':last_name', $data['last_name'], PDO::PARAM_STR);
+        $pstmt->execute();
+        $data = $pstmt->fetchAll();
+        if (empty($data)) {
+            throw new Exception("Student record not found.");
+        }
+        foreach ($data as $row) {
+            echo $row['student_id'].$row['first_name'].$row['last_name'].$row['birth_date'].$row['gender']. ' ';
+        }
+    } catch (Exception $e) {
+        echo $e->getMessage();
+    }
+?>
+```
+/// type=SS, answer=[5]
+
+Execute the program. What is its output?
+
+- It produces an error.
+
+- No output is displayed.
+
+- It prints `Unable to establish a connection.`.
+
+- It prints `Successfully connected to the database.`.
+
+- It prints `Successfully connected to the database.` and `Student record not found.`.
+
+
+/// type=SS, answer=[4]
+
+On lines 5 and 6 of `student.php`, replace the strings `John` and `Smith` with `Alisa` and `Ells`. Execute the program. What is its output?
+
+- It produces an error.
+
+- No output is displayed.
+
+- It prints `Unable to establish a connection.`.
+
+- It prints `Successfully connected to the database.` and `6AlisaElls1999-06-30F`.
+
+- It prints `Successfully connected to the database.` and `Student record not found.`.
+
+
+/// type=SS, answer=[4]
+
+On lines 13 and 14 of `student.php`, replace the `bindValue()` method with `bindParam()`. Execute the program. What is its output?
```

Comment:
Are you expecting to have the values in the question on line 607 to reflect in the code here? If yes, then you have to add a new REPL before this question. Please check. Thanks.